README.md
<div align='center'>
  
  ![npm](https://badgen.net/badge/license/MIT/blue)
  [![npm](https://badgen.net/npm/dt/react-native-reanimated-switch)](https://www.npmjs.com/package/react-native-reanimated-switch)
  [![npm](https://badgen.net/npm/v/react-native-reanimated-switch)](https://www.npmjs.com/package/react-native-reanimated-switch)

  <h1>React Native Reanimated Switch</h1>
  <img width="auto" height="550" src="./example/RNSwitchExample/assets/ios.gif">
  <img width="auto" height="550" src="./example/RNSwitchExample/assets/android.gif">

</div>

## :anchor: Installation

```sh

yarn add react-native-reanimated-switch
# or
npm i react-native-reanimated-switch

```

## :family: Dependencies

##### React Native Reanimated

```sh
npm install react-native-reanimated
```

For iOS

```sh

cd ios && pod install && cd ..

```

> For detailed instructions check it out [here](https://docs.swmansion.com/react-native-reanimated/docs/next/installation)

> Rebuild the project after adding the dependencies

## :mag: Usage

```js
import React, { useState } from "react";
import { SafeAreaView, StatusBar, StyleSheet, Text, View } from "react-native";
import RNSwitch from "./src/Switch";

const App = () => {
  const [switchState, setSwitchState] = useState(false);
  const handleOnPressSwitch = (value) => {
    setSwitchState(value);
  };
  return (
    <>
      <StatusBar
        barStyle="dark-content"
        translucent={true}
        backgroundColor="white"
      />
      <SafeAreaView style={styles.container}>
        <Text style={styles.header}>React Native Reanimated Switch</Text>
        <Text style={styles.header}>Default </Text>
        <RNSwitch handleOnPress={handleOnPressSwitch} value={switchState} />
        <Text style={styles.header}>Color Customisable </Text>
        <View style={styles.switchContainer}>
          <RNSwitch
            handleOnPress={handleOnPressSwitch}
            activeTrackColor="#FE5F8F"
            value={switchState}
          />
          <RNSwitch
            handleOnPress={handleOnPressSwitch}
            activeTrackColor="#667eea"
            value={switchState}
          />
          <RNSwitch
            handleOnPress={handleOnPressSwitch}
            activeTrackColor="#ed8936"
            value={switchState}
          />
          <RNSwitch
            handleOnPress={handleOnPressSwitch}
            activeTrackColor="#feb2b2"
            value={switchState}
          />
        </View>
      </SafeAreaView>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: "flex",
    alignContent: "center",
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "white",
  },
  header: {
    fontSize: 24,
    paddingVertical: 30,
  },
  switchContainer: {
    display: "flex",
    justifyContent: "space-between",
    flexDirection: "row",
    width: "90%",
  },
});

export default App;
```

## :wrench: Props

| Name               | Description                              | Required | Type     | Default |
| ------------------ | ---------------------------------------- | -------- | -------- | ------- |
| value              | State of switch component                | YES      | Boolean  | -       |
| handleOnPress      | A callback with the current switch state | YES      | Function | -       |
| activeTrackColor   | The track color when switch is active    | NO       | Color    | #007AFF |
| inActiveTrackColor | The track color when switch is inactive  | NO       | Color    | #F2F5F7 |
| thumbColor         | The color of switch thumb                | NO       | Color    | #FFFFFF |

## :tada: Example

Checkout the example [here](https://github.com/timelessco/react-native-reanimated-switch/tree/master/example/RNSearch).

## :notebook: Blog

Read my blog [here](https://medium.com/timeless/react-native-reanimated-switch-83c331af7877?source=friends_link&sk=7155a8907037c741e717d081a4a29498)

## :snowman: Built with ❤️ and

- [react-native](https://www.npmjs.com/package/react-native)
- [react-native-reanimated](https://docs.swmansion.com/react-native-reanimated/)

## :v: Contributing

Pull requests are always welcome! Feel free to open a new GitHub issue for any changes that can be made.

## :man: Author

[Karthik B](https://twitter.com/_iam_karthik)

## :clipboard: License

MIT


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### 0.0.3 (2020-08-30)


### Features

* :alien: Added assets to example project ([2dc6d1e](https://github.com/timelessco/react-native-reanimated-switch/commit/2dc6d1e38548e898886addfceb1bf021d5f73811))
* :fire: Updated Example Project ([6a0e016](https://github.com/timelessco/react-native-reanimated-switch/commit/6a0e016dd9119364c22cc6190125e4073af299c3))
* :pencil: Added README.md ([73dbddc](https://github.com/timelessco/react-native-reanimated-switch/commit/73dbddcc30b27214f476985c80a8875330d05849))
* :pencil: Updated README.md ([5a71e1f](https://github.com/timelessco/react-native-reanimated-switch/commit/5a71e1f622402fa12474dd002943e41c8ddffc97))
* :sparkles: Added Example app to try Switch Component ([d2a563c](https://github.com/timelessco/react-native-reanimated-switch/commit/d2a563c6199fc2628a46c2954494141c16858182))
* :tada: Initial release v0.0.1 ([6d5b557](https://github.com/timelessco/react-native-reanimated-switch/commit/6d5b557486b68aad0d358c6d90381953a3c9b562))
* :zap: Added memoized callback and value prop to switch component ([50153a5](https://github.com/timelessco/react-native-reanimated-switch/commit/50153a5841faf9d32eaa623669b0b913976ef85c))


### Bug Fixes

* :bug: Added isRequired for value prop ([06338fc](https://github.com/timelessco/react-native-reanimated-switch/commit/06338fcb53d281e4ca8de8ff3bcbd1934d2362e7))
* :bug: Fixed a typo error ([1bffd07](https://github.com/timelessco/react-native-reanimated-switch/commit/1bffd0776619ba76ce4ce4975b985dc502c23bb5))
* :lock: updated peer dependencies ([044565b](https://github.com/timelessco/react-native-reanimated-switch/commit/044565b9b13a3535e9b7607674e6b274a80a2aa0))

### 0.0.2 (2020-08-30)


### Features

* :alien: Added assets to example project ([2dc6d1e](https://github.com/timelessco/react-native-reanimated-switch/commit/2dc6d1e38548e898886addfceb1bf021d5f73811))
* :fire: Updated Example Project ([6a0e016](https://github.com/timelessco/react-native-reanimated-switch/commit/6a0e016dd9119364c22cc6190125e4073af299c3))
* :pencil: Added README.md ([73dbddc](https://github.com/timelessco/react-native-reanimated-switch/commit/73dbddcc30b27214f476985c80a8875330d05849))
* :pencil: Updated README.md ([5a71e1f](https://github.com/timelessco/react-native-reanimated-switch/commit/5a71e1f622402fa12474dd002943e41c8ddffc97))
* :sparkles: Added Example app to try Switch Component ([d2a563c](https://github.com/timelessco/react-native-reanimated-switch/commit/d2a563c6199fc2628a46c2954494141c16858182))
* :tada: Initial release v0.0.1 ([6d5b557](https://github.com/timelessco/react-native-reanimated-switch/commit/6d5b557486b68aad0d358c6d90381953a3c9b562))
* :zap: Added memoized callback and value prop to switch component ([50153a5](https://github.com/timelessco/react-native-reanimated-switch/commit/50153a5841faf9d32eaa623669b0b913976ef85c))


### Bug Fixes

* :bug: Added isRequired for value prop ([06338fc](https://github.com/timelessco/react-native-reanimated-switch/commit/06338fcb53d281e4ca8de8ff3bcbd1934d2362e7))
* :lock: updated peer dependencies ([044565b](https://github.com/timelessco/react-native-reanimated-switch/commit/044565b9b13a3535e9b7607674e6b274a80a2aa0))

### 0.0.1 (2020-08-29)

### Features

Basic React Native Switch Component


example/RNSwitchExample/App.js
import React, {useState} from 'react';
import {SafeAreaView, StatusBar, StyleSheet, Text, View} from 'react-native';
import RNSwitch from 'react-native-reanimated-switch';

const App = () => {
  const [switchState, setSwitchState] = useState(false);
  const handleOnPressSwitch = (value) => {
    setSwitchState(value);
  };
  return (
    <>
      <StatusBar
        barStyle="dark-content"
        translucent={true}
        backgroundColor="white"
      />
      <SafeAreaView style={styles.container}>
        <Text style={styles.header}>React Native Reanimated Switch</Text>
        <Text style={styles.header}>Default </Text>
        <RNSwitch handleOnPress={handleOnPressSwitch} value={switchState} />
        <Text style={styles.header}>Color Customisable </Text>
        <View style={styles.switchContainer}>
          <RNSwitch
            handleOnPress={handleOnPressSwitch}
            activeTrackColor="#FE5F8F"
            value={switchState}
          />
          <RNSwitch
            handleOnPress={handleOnPressSwitch}
            activeTrackColor="#667eea"
            value={switchState}
          />
          <RNSwitch
            handleOnPress={handleOnPressSwitch}
            activeTrackColor="#ed8936"
            value={switchState}
          />
          <RNSwitch
            handleOnPress={handleOnPressSwitch}
            activeTrackColor="#feb2b2"
            value={switchState}
          />
        </View>
      </SafeAreaView>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: 'flex',
    alignContent: 'center',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
  },
  header: {
    fontSize: 24,
    paddingVertical: 30,
  },
  switchContainer: {
    display: 'flex',
    justifyContent: 'space-between',
    flexDirection: 'row',
    width: '90%',
  },
});

export default App;
